apps/mac.c

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+
+#include "apps.h"
+#include "progs.h"
+#include <openssl/bio.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+
+#undef BUFSIZE
+#define BUFSIZE 1024*8
+
+typedef enum OPTION_choice {
+    OPT_ERR = -1, OPT_EOF = 0, OPT_HELP,
+    OPT_MAC, OPT_MACOPT, OPT_BIN, OPT_OUT
+} OPTION_CHOICE;
+
+const OPTIONS mac_options[] = {
+    {OPT_HELP_STR, 1, '-', "Usage: %s [options] [file]\n"},
+    {OPT_HELP_STR, 1, '-', "  file\t\tto mac (default is stdin)\n"},
+    {"help", OPT_HELP, '-', "Display this summary"},
+    {"name", OPT_MAC, 's', "MAC algorithm (See list -mac-algorithms)"},
```

Comment:
For consistency, you could choose the same type of option as `enc` and `dgst`, i.e. to make the mac name an actual option itself.  `-hmac`, `-gmac`, `-cmac`...

Another option would be to use the same option name as `genpkey`.

Reply:
Yet another option, since this is mandatory anyway, is to make the name an argument instead of an option value.  Personally, I would prefer this, as it seems quite pointless to use an option -- i.e. something that's supposed to be *optional* -- for something that's actually mandatory...  but that does break consistency with a number of other newer commands, so it's not as easy to argue for when I'm talking about consistency.  However, this is a breach of consistency I would personally encourage.

(I am considering submitting a PR that give other commands the possibility to take mandatory arguments as arguments instead of options, if those options haven't been given ...  that last bit is for backward compatibility)